numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -2334,13 +2334,14 @@ array_fromstring(PyObject *NPY_UNUSED(ignored), PyObject *args, PyObject *keywds
     Py_ssize_t nin = -1;
     char *sep = NULL;
     Py_ssize_t s;
-    static char *kwlist[] = {"string", "dtype", "count", "sep", "like", NULL};
+    static char *kwlist[] = {"string", "dtype", "count", "sep", "like", "ignore_whitespace", NULL};
     PyObject *like = Py_None;
     PyArray_Descr *descr = NULL;
+    int ignore_whitespace = 0; // default behavior is to not ignore whitespace
 
     if (!PyArg_ParseTupleAndKeywords(args, keywds,
-                "s#|O&" NPY_SSIZE_T_PYFMT "s$O:fromstring", kwlist,
-                &data, &s, PyArray_DescrConverter, &descr, &nin, &sep, &like)) {
+                "s#|O&" NPY_SSIZE_T_PYFMT "s$O$p:fromstring", kwlist,
```

Comment:
The tests are failing because of this addition. See https://docs.python.org/3/c-api/arg.html#strings-and-buffers for documentation in Python for how to write these format strings. As that document says, `$` indicates that the rest of the arguments are keyword-only arguments and can't be included twice.

I'd also encourage you to build and test your changes to numpy locally. See https://numpy.org/devdocs/dev/development_environment.html for more details about how to run the numpy tests in your local development environment.

Reply:
im running tests locally, but i have had some trouble with the setup. `spin build` and `spin test` run tests successfully while `python runtests.py -v ` returns some failed tests. I opened the PR to start a conversation. Thanks for the feedback! I'll get around it in a day or 2.